python/pyarrow/public-api.pxi

```diff
@@ -50,6 +79,19 @@ cdef public api object pyarrow_wrap_data_type(
     return out
 
 
+cdef public api bint pyarrow_is_field(object field):
+    return isinstance(field, Field)
+
+
+cdef public api shared_ptr[CField] pyarrow_unwrap_field(object field):
+    cdef Field field_
+    if pyarrow_is_field(field):
+        field_ = <Field>(field)
+        return field_.sp_field
+
+    return shared_ptr[CField]()
```

Comment:
Yes, nullptr would only invoke the default constructor and return the same object.